flair/embeddings.py

```diff
@@ -542,6 +544,76 @@ def extra_repr(self):
         return "min_freq={}".format(self.min_freq)
 
 
+class HashEmbeddings(TokenEmbeddings):
+    """Standard embeddings with Hashing Trick."""
+
+    def __init__(
+        self,
+        num_embeddings: int = 1000,
+        embedding_length: int = 300,
+        hash_method='md5'
+    ):
+
+        super().__init__()
+        self.name = "hash"
+        self.static_embeddings = False
+
+        self.__num_embeddings = num_embeddings
+        self.__embedding_length = embedding_length
+
+        self.__hash_method = hash_method
+
+        # model architecture
+        self.embedding_layer = torch.nn.Embedding(
+            self.__num_embeddings, self.__embedding_length
+        )
+        torch.nn.init.xavier_uniform_(self.embedding_layer.weight)
+
+        self.to(flair.device)
+
+
+    @property
+    def num_embeddings(self) -> int:
+        return self.__num_embeddings
+
+    @property
+    def embedding_length(self) -> int:
+        return self.__embedding_length
+
+    def _add_embeddings_internal(self, sentences: List[Sentence]) -> List[Sentence]:
+
+        def get_idx_for_item(text):
+            hash_function = hashlib.new(self.__hash_method)
+            hash_function.update(bytes(str(text), 'utf-8'))
+            return int(self.hash_function.hexdigest(), 16) % self.__num_embeddings
```

Comment:
`self.hash_function` throws an error since `has_function` is no longer a global variable. This should be replaced with: 
`return int(hash_function.hexdigest(), 16) % self.__num_embeddings`